### PR TITLE
Support [run] entry in beamtalk.toml for REPL startup entry point (BT-740)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

- Adds `[run]` section to `beamtalk.toml` with an `entry` key (e.g. `entry = "Main run"`)
- `beamtalk new` generates `beamtalk.toml` with `[run]` section defaulting to `entry = "Main run"`
- `beamtalk repl` evaluates the entry expression after auto-loading compiled classes; prints the expression and result; warns on failure but always starts the REPL interactively
- Backwards compatible: no `[run]` section means normal startup

Linear: https://linear.app/beamtalk/issue/BT-740

## Changes

- `crates/beamtalk-cli/src/commands/manifest.rs`: Added `RunConfig` struct, `run: Option<RunConfig>` field to `Manifest`, and `find_run_config()` function with unit tests
- `crates/beamtalk-cli/src/commands/repl/mod.rs`: Added `auto_run_entry()` helper called after REPL connects, before the interactive loop
- `crates/beamtalk-cli/src/commands/new.rs`: Added `[run]` section with `entry = "Main run"` to the project template

## Test plan

- [x] Unit tests for `RunConfig` parsing and `find_run_config()` edge cases
- [x] Template test updated to assert `[run]` section and `entry = "Main run"` present
- [x] Full CI passes: `just ci`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added support for a [run] entry in beamtalk.toml and added it to the default project template; the REPL will auto-execute the configured entry on startup and after reconnect/listing actors. Failures emit warnings and keep the REPL interactive, printing the entry and its result.

* **Tests**
  * Added/updated tests covering presence, absence, no-manifest, invalid package name, and custom values for the [run] configuration.

### CODE REVIEW

File: crates/beamtalk-cli/src/commands/manifest.rs [hunk 1]
```diff
@@ -84,6 +97,28 @@ pub fn find_manifest(project_root: &Utf8Path) -> Result<Option<PackageManifest>>
...
+    {
+        return Ok(None);
+    }
+    let content = fs::read_to_string(&manifest_path)
+        .into_diagnostic()
+        .wrap_err_with(|| format!("Failed to read manifest '{manifest_path}'"))?;
+    let manifest: Manifest = toml::from_str(&content)
+        .into_diagnostic()
+        .wrap_err_with(|| format!("Failed to parse manifest '{manifest_path}'"))?;
+    Ok(manifest.run)
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-cli/src/commands/manifest.rs]: Fixed in 2708a00: extracted a private `parse_manifest_raw()` helper that reads and deserializes the TOML without package-name validation. Both `parse_manifest()` and `find_run_config()` now delegate to this shared helper, eliminating the duplication.
